client/src/components/user/dashboard.js

```diff
@@ -139,40 +178,77 @@ class Dashboard extends React.Component {
       );
     }
 
-    let display;
-    if (activeSubscription) {
-     display = <SubscriptionDetail setActiveSubscription={this.setActiveSubscription} setEditingSubscription={this.setEditingSubscription} handleDelete={this.handleDelete} details={activeSubscription} />;
-    } else if (editingSubscription || addingSubscription) {
-      display = (
-        <div className="col-12">
-          <div className="d-none d-md-block">
-            {subscriptionForm}
+    switch (mainComponentView) {
+      case 'subscriptionDetail':
+        return <SubscriptionDetail setActiveSubscription={this.setActiveSubscription} handleDashboard={this.handleDashboardChange} handleDelete={this.handleDelete} details={activeSubscription} />;
+      case 'createSubscription':
+        return (
+          <div>
+            <div className="p-3 m-2 d-flex flex-wrap borderSubscriptionForm d-none d-md-block">
+              <div className="col d-flex justify-content-between align-items-center">
+                <div />
+                <h2 className="text-start">Create Subscription</h2>
+                <button className="btn btn-link my-2 d-md-none" type="button" data-bs-dismiss="offcanvas" aria-label="Close" onClick={() => this.handleDashboardChange('dashboardCalendar')}>
+                  <FontAwesomeIcon icon={faTimes} />
+                </button>
+                <button className="btn btn-link my-2 d-none d-md-block" type="button" onClick={() => this.handleDashboardChange('dashboardCalendar')}>
+                  <FontAwesomeIcon icon={faTimes} />
+                </button>
+              </div>
+              <CreateSubscription 
+                addSubscription={this.handleUpdate}
+                toggleLoadingState={this.toggleLoadingState}
+                showSubscriptionList={this.showSubscriptionList}
+                currentSubscriptions={subscriptions} />
+            </div>
+            <div className="d-md-none">
+              <DashboardCalendar subscriptions={subscriptions} />
+            </div>
           </div>
-          <div className="d-md-none">
-            <DashboardCalendar subscriptions={subscriptions}/>
+        );
+      case 'updateSubscription': 
+        return (
+          <div>
+            <div className="p-3 m-2 d-flex flex-wrap borderSubscriptionForm d-none d-md-block">
+              <div className="col d-flex justify-content-between align-items-center">
+                <div />
```

Comment:
what's going on with this div?

Reply:
the parent div is a Flexbox container, and the empty div places empty space in the first spot, resulting in the title in the middle and the close button at the end of the row.